__bass.py

```diff
@@ -10,24 +10,24 @@
 from __future__ import print_function
 
 import json
-import os
 import subprocess
 import sys
 import tempfile
 
 
-def gen_script():
-    fd, name = tempfile.mkstemp()
+BASH = 'bash'
+
 
+def gen_script():
     divider = '-__-__-__bass___-env-output-__bass_-__-__-__-__'
 
-    old_env = os.popen('/bin/bash -c "env"', 'r').read().splitlines()
+    args = [BASH, '-c', 'env']
+    output = subprocess.check_output(args, universal_newlines=True)
+    old_env = output.splitlines()
 
     command = '{}; echo "{}"; env'.format(' '.join(sys.argv[1:]), divider)
-    output = subprocess.check_output(['bash', '-c', command])
-    if bytes != str and isinstance(output, bytes):
-        # on python 3
-        output = str(output, 'utf-8')
+    args = [BASH, '-c', command]
+    output = subprocess.check_output(args, universal_newlines=True)
```

Comment:
Hi Eddie,
Thank you very much for your quick feedback. Adding the `universal_newlines=True` parameter to `subprocess.check_output` automatically decodes the result to a `str` as you can see below, so it should make the conversion to string you had redundant, no?
```python
Python 3.4.3 (default, Mar 25 2015, 17:13:50) 
[GCC 4.9.2 20150304 (prerelease)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import subprocess
>>> type(subprocess.check_output('ls', universal_newlines=True))
<class 'str'>
```